Tools/cases_generator/analyzer.py

```diff
@@ -0,0 +1,410 @@
+
+from dataclasses import dataclass
+import lexer
+import parser
+from _typing_backports import assert_never
+
+@dataclass
+class Properties:
+    escapes: bool
+    infallible: bool
+    deopts: bool
+    oparg: bool
+    jumps: bool
+    ends_with_eval_breaker: bool
+    needs_this: bool
+    always_exits: bool
+    stores_sp: bool
+
+    def dump(self, indent):
+        print(indent,
+              "escapes:", self.escapes,
+              ", infallible:", self.infallible,
+              ", deopts:", self.deopts,
+              ", oparg:", self.oparg,
+              ", jumps:", self.jumps,
+              ", eval_breaker:", self.ends_with_eval_breaker,
+              ", needs_this:", self.needs_this,
+              ", always_exits:", self.always_exits,
+              ", stores_sp:", self.stores_sp,
+              sep=""
+        )
+
+
+SKIP_PROPERTIES = Properties(
```

Comment:
I'm sure dataclasses let you use keyword args and defaults here. :-)

Reply:
And a ``__str__`` (so `dump()` might not be needed))